### PR TITLE
Js preflight only

### PR DESCRIPTION
**Pull request by jsingle**
@thieman

### CODE REVIEW

File: dusty/preflight.py [hunk 1]
```diff
@@ -16,6 +16,15 @@
...
     pass
 
+def returns_exception(f):
+    def inner():
+        try:
+            f()
+            return None
+        except Exception as e:
+            return e
+    return inner
```
Comment (thieman) [inline dusty/preflight.py]: `if err != nil`

File: dusty/preflight.py [hunk 2]
```diff
@@ -73,14 +89,18 @@ def _ensure_config_dir_exists():
...
 def preflight_check():
     logging.info('Starting preflight check')
+    errors = []
+    errors.append(_check_nginx())
+    errors.append(_check_rsync())
+    errors.append(_check_virtualbox())
+    errors.append(_check_boot2docker())
+    errors.append(_check_docker())
+    errors.append(_check_docker_compose())
+    errors.append(_assert_hosts_file_is_writable())
```
Comment (thieman) [inline dusty/preflight.py]: Get thunky with it

```
for check in [_check_nginx, ...]:
    errors.append(check())
```

You can take advantage of them all having the same call signature.

Comment (thieman) [inline dusty/preflight.py]: Shit you can even just do a list comprehension like a boss

Comment (jsingle) [inline dusty/preflight.py]: sounds better

**Comment by thieman** [comment]
Cool PR bro

**Pull request merged**